### PR TITLE
ci: auto generate natspec docs

### LINKED ISSUE #739: Autogenerate natspec docs
There are [natspec docs](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/docs/src/SUMMARY.md) generated locally by this [command](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/package.json#L38). We should automate this flow via CI.

What should be done:
- Create a CI workflow (or add an additional step to the `Build and Test` workflow) which on every push to the `development` branch will regenerate natspec docs

### PR DESCRIPTION
**Pull request by tungbq**
Resolves #739

### CODE REVIEW

**Comment by tungbq** [comment]
This was tested on my forked repo. Please check the result here:
https://github.com/tungbq/ubiquity-dollar/actions/runs/5725286592/job/15513481521

File: .github/workflows/generate-doc.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Auto Generate Doc
+
+on:
```
Comment (rndquu) [inline .github/workflows/generate-doc.yml:3]: pls add the `workflow_dispatch` handler

File: .github/workflows/generate-doc.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18.14.1"
+
+      - name: Install dependencies
```
Comment (rndquu) [inline .github/workflows/generate-doc.yml]: unnecessary step, pls remove

File: .github/workflows/generate-doc.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+      - name: Setup Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly-cc5637a979050c39b3d06bc4cc6134f0591ee8d0
+
+      - name: Yarn Install
+        run: yarn install --mode=skip-build && yarn allow-scripts
+        
+      - name: Setup Forge
```
Comment (rndquu) [inline .github/workflows/generate-doc.yml]: unnecessary step, pls remove

**Review by rndquu — CHANGES_REQUESTED** [review]

File: .github/workflows/generate-doc.yml (hunk 1 above)
Comment (tungbq) [reply .github/workflows/generate-doc.yml:3]: Added

File: .github/workflows/generate-doc.yml (hunk 2 above)
Comment (tungbq) [reply .github/workflows/generate-doc.yml]: Thanks. Removed it

File: .github/workflows/generate-doc.yml (hunk 3 above)
Comment (tungbq) [reply .github/workflows/generate-doc.yml]: Thanks. Removed it

**Review by rndquu — APPROVED** [review]

**Review by molecula451 — APPROVED** [review]

**Comment by molecula451** [comment]
thanks for contributing @tungbq, good work, working good

**Pull request merged**